src/windows/SliderWindows/SliderView.cpp

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+
+#include "JSValueXaml.h"
+
+#include "SliderView.h"
+#include "SliderView.g.cpp"
+
+#include "winrt/Windows.UI.Xaml.Media.h"
+#include "winrt/Windows.UI.Xaml.Media.Imaging.h"
+#include "winrt/Windows.UI.Xaml.Input.h"
+
+namespace winrt {
+    using namespace Microsoft::ReactNative;
+    using namespace Windows::Foundation;
+    using namespace Windows::UI;
+    using namespace Windows::UI::Xaml::Media;
+    using namespace Windows::UI::Xaml::Media::Imaging;
+}
+
+namespace winrt::SliderWindows::implementation {
+
+    SliderView::SliderView(winrt::IReactContext const& reactContext) : m_reactContext(reactContext) {
+        RegisterEvents();
+    }
+
+    void SliderView::RegisterEvents() {
+        //Set this so that Slider passes on Manipulation* events.
+        this->ManipulationMode(xaml::Input::ManipulationModes::All);
+
+        m_sliderValueChangedRevoker = this->ValueChanged(winrt::auto_revoke,
+            [ref = get_weak()](auto const& sender, auto const& args) {
+            if (auto self = ref.get()) {
+                self->OnValueChangedHandler(sender, args);
+            }
+        });
+
+        m_sliderManipulationStartingRevoker = this->ManipulationStarting(winrt::auto_revoke,
+            [ref = get_weak()](auto const& sender, auto const& args) {
+            if (auto self = ref.get()) {
+                self->OnManipulationStartingHandler(sender, args);
+            }
+        });
+
+        m_sliderManipulationCompletedRevoker = this->ManipulationCompleted(winrt::auto_revoke,
+            [ref = get_weak()](auto const& sender, auto const& args) {
+            if (auto self = ref.get()) {
+                self->OnManipulationCompletedHandler(sender, args);
+            }
+        });
+    }
+
+    void SliderView::UpdateProperties(winrt::IJSValueReader const& reader) {
+        m_updating = true;
+
+        bool updatedValue = false;
+        bool updatedMaxValue = false;
+        bool updatedMinValue = false;
+
+        auto const& propertyMap = JSValueObject::ReadFrom(reader);
+
+        for (auto const& pair : propertyMap) {
+            auto const& propertyName = pair.first;
+            auto const& propertyValue = pair.second;
+
+            if (propertyName == "value") {
+                if (propertyValue.IsNull()) {
+                    this->ClearValue(xaml::Controls::Primitives::RangeBase::ValueProperty());
+                }
+                else {
```

Comment:
is this using clang format? seems like the formatting is different than in our repo?

Reply:
I just used formatting that I find clearer, with more spacing between if-else clauses. It's not our repo, so I didn't know if we want to impose the same style

Reply:
The same formatting was used in the ProgressView and Picker repos when adding Windows support.